commit 642ba016ef55d0f16cdca89307cfacfff3b64889
Author: adlinkshiva <shivashankar.thiruchitrambalam@adlinktech.com>
Date:   Sun Mar 31 01:06:40 2024 +0530

    Build_support_CAN&SPI

diff --git a/make/target/product/base_system.mk b/make/target/product/base_system.mk
index b0870c3..0169190 100644
--- a/make/target/product/base_system.mk
+++ b/make/target/product/base_system.mk
@@ -376,6 +376,8 @@ PRODUCT_PACKAGES_DEBUG := \
     idlcli \
     init-debug.rc \
     iotop \
+    candump \
+    cansend \
     iperf3 \
     iw \
     logpersist.start \
@@ -395,6 +397,7 @@ PRODUCT_PACKAGES_DEBUG := \
     tracepath \
     tracepath6 \
     traceroute6 \
+    spidevtest \
     unwind_info \
     unwind_reg_info \
     unwind_symbols \
