commit 4b97818e1603722308879c410e96dc21d829d98e
Author: shivaadlink <shivashankar.thiruchitrambalam@adlinktech.com>
Date:   Thu Mar 21 10:55:02 2024 +0530

    0003-Single_image_2g_hdmi_cmafix

diff --git a/imx8m/sp2_imx8mp/BoardConfig.mk b/imx8m/sp2_imx8mp/BoardConfig.mk
index 92ba9b28..549e965e 100644
--- a/imx8m/sp2_imx8mp/BoardConfig.mk
+++ b/imx8m/sp2_imx8mp/BoardConfig.mk
@@ -129,7 +129,7 @@ BOARD_USE_SENSOR_FUSION := true
 # -------@block_kernel_bootimg-------
 BOARD_KERNEL_BASE := 0x40400000
 
-CMASIZE=1184M
+CMASIZE=960M
 # NXP default config
 BOARD_KERNEL_CMDLINE := init=/init firmware_class.path=/vendor/firmware loop.max_part=7 bootconfig
 BOARD_BOOTCONFIG += androidboot.console=ttymxc1 androidboot.hardware=nxp
@@ -147,10 +147,10 @@ BOARD_KERNEL_CMDLINE +=  moal.mod_para=wifi_mod_para.conf pci=nomsi
 
 # low memory device build config
 ifeq ($(LOW_MEMORY),true)
-BOARD_KERNEL_CMDLINE += cma=320M@0x400M-0xb80M galcore.contiguousSize=33554432
+BOARD_KERNEL_CMDLINE += cma=128M@0x400M-0x800M galcore.contiguousSize=33554432
 BOARD_BOOTCONFIG += androidboot.displaymode=720p
 else
-BOARD_KERNEL_CMDLINE += cma=$(CMASIZE)@0x400M-0x1000M
+BOARD_KERNEL_CMDLINE += cma=800M@0x400M-0xb80M
 endif
 
 # powersave config
diff --git a/imx8m/sp2_imx8mp/UbootKernelBoardConfig.mk b/imx8m/sp2_imx8mp/UbootKernelBoardConfig.mk
index 0c5d81f4..94f5821f 100644
--- a/imx8m/sp2_imx8mp/UbootKernelBoardConfig.mk
+++ b/imx8m/sp2_imx8mp/UbootKernelBoardConfig.mk
@@ -3,15 +3,15 @@ TARGET_BOOTLOADER_POSTFIX := bin
 UBOOT_POST_PROCESS := true
 
 # u-boot target
-TARGET_BOOTLOADER_CONFIG := imx8mp:imx8mp_evk_android_defconfig
-TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-secure-unlock-dual:imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
-TARGET_BOOTLOADER_CONFIG += imx8mp-dual:imx8mp_evk_android_dual_defconfig
-TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-dual:imx8mp_evk_android_trusty_dual_defconfig
+TARGET_BOOTLOADER_CONFIG := imx8mp:imx8mp_sp2_android_defconfig
+#TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-secure-unlock-dual:imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
+#TARGET_BOOTLOADER_CONFIG += imx8mp-dual:imx8mp_evk_android_dual_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-dual:imx8mp_sp2_android_trusty_dual_defconfig
 ifeq ($(POWERSAVE),true)
-TARGET_BOOTLOADER_CONFIG += imx8mp-powersave:imx8mp_evk_android_powersave_defconfig
-TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-powersave-dual:imx8mp_evk_android_trusty_powersave_dual_defconfig
+#TARGET_BOOTLOADER_CONFIG += imx8mp-powersave:imx8mp_evk_android_powersave_defconfig
+#TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-powersave-dual:imx8mp_evk_android_trusty_powersave_dual_defconfig
 endif
-TARGET_BOOTLOADER_CONFIG += imx8mp-evk-uuu:imx8mp_evk_android_uuu_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-evk-uuu:imx8mp_sp2_android_uuu_defconfig
 
 ifeq ($(IMX8MP_USES_GKI),true)
 TARGET_KERNEL_DEFCONFIG := gki_defconfig
