diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index dfc906d8..299d4ba4 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -1,4 +1,6 @@
 #!/bin/bash
+export AARCH64_GCC_CROSS_COMPILE=/opt/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu/bin/aarch64-linux-gnu-
+export CLANG_PATH=/opt/prebuilt-android-clang/
 
 # help function, it display the usage of this script.
 help() {
@@ -131,6 +133,12 @@ if [ -n "${build_kernel}" ] && [ ${TARGET_PRODUCT} = "evk_8mp" ]; then
     build_kernel_module_flag=1;
 fi
 
+# vvcam.ko need build with kernel each time to make sure "insmod vvcam.ko" works
+if [ -n "${build_kernel}" ] && [ ${TARGET_PRODUCT} = "lec_imx8mp" ]; then
+    build_vvcam="vvcam";
+    build_kernel_module_flag=1;
+fi
+
 # mlan.ko and moal.ko need build with kernel each time to make sure "insmod mlan.ko" and "insmod moal.ko" works
 if [ -n "${build_kernel}" ]; then
     build_mxmwifi="mxmwifi";
diff --git a/imx8m/AndroidProducts.mk b/imx8m/AndroidProducts.mk
index 924d44fe..cf69f6b5 100644
--- a/imx8m/AndroidProducts.mk
+++ b/imx8m/AndroidProducts.mk
@@ -32,7 +32,8 @@ PRODUCT_MAKEFILES := \
   $(LOCAL_DIR)/evk_8mm/evk_8mm.mk \
   $(LOCAL_DIR)/evk_8mm/evk_8mm_drm.mk \
   $(LOCAL_DIR)/evk_8mn/evk_8mn.mk \
-  $(LOCAL_DIR)/evk_8mp/evk_8mp.mk
+  $(LOCAL_DIR)/evk_8mp/evk_8mp.mk \
+  $(LOCAL_DIR)/lec_imx8mp/lec_imx8mp.mk
 
 COMMON_LUNCH_CHOICES := \
   evk_8mq-user \
@@ -46,4 +47,6 @@ COMMON_LUNCH_CHOICES := \
   evk_8mn-user \
   evk_8mn-userdebug \
   evk_8mp-user \
-  evk_8mp-userdebug
+  evk_8mp-userdebug \
+  lec_imx8mp-user \
+  lec_imx8mp-userdebug
