commit 8ad8269a3d1c28144cf66701454e52cb589b0af7
Author: Veera H <veera.h@adlinktech.com>
Date:   Tue Aug 30 20:46:55 2022 +0530

    bluetooth driver configuration

diff --git a/imx8m/lec_imx8mp/BoardConfig.mk b/imx8m/lec_imx8mp/BoardConfig.mk
index 91436dea..169728a9 100644
--- a/imx8m/lec_imx8mp/BoardConfig.mk
+++ b/imx8m/lec_imx8mp/BoardConfig.mk
@@ -134,6 +134,8 @@ else
     BOARD_KERNEL_CMDLINE += androidboot.wificountrycode=CN moal.mod_para=wifi_mod_para.conf moal.sta_name=wlan moal.uap_name=wlan moal.wfd_name=p2p
 endif
 
+BOARD_KERNEL_CMDLINE += androidboot.bt_uart=/dev/mbtchar0 androidboot.btmacaddr=2f:0b:a1:aa:bf:04
+
 # low memory device build config
 ifeq ($(LOW_MEMORY),true)
 BOARD_KERNEL_CMDLINE += cma=320M@0x400M-0xb80M androidboot.displaymode=720p galcore.contiguousSize=33554432
diff --git a/imx8m/lec_imx8mp/sepolicy/file_contexts b/imx8m/lec_imx8mp/sepolicy/file_contexts
index ee3f218c..c3a53c60 100644
--- a/imx8m/lec_imx8mp/sepolicy/file_contexts
+++ b/imx8m/lec_imx8mp/sepolicy/file_contexts
@@ -4,6 +4,8 @@
 #peripheral
 /vendor/bin/fsl_sensor_fusion   u:object_r:sensors_exec:s0
 /dev/ttymxc[0-9]*               u:object_r:tty_device:s0
+/dev/mbtchar0               	u:object_r:tty_device:s0
+/dev/mdebugchar0		u:object_r:tty_device:s0
 /dev/sda[0-8]*                  u:object_r:fuse:s0
 
 #Thermal HAL service
diff --git a/imx8m/lec_imx8mp/ueventd.nxp.rc b/imx8m/lec_imx8mp/ueventd.nxp.rc
index a9b0d081..0630e34a 100644
--- a/imx8m/lec_imx8mp/ueventd.nxp.rc
+++ b/imx8m/lec_imx8mp/ueventd.nxp.rc
@@ -1,5 +1,6 @@
 /dev/block/by-name/presistdata 0600   system     system
 /dev/ttymxc0              0660   bluetooth  bluetooth
+/dev/mbtchar0             0660   bluetooth  bluetooth
 /dev/snd/*                0660   system     audio
 /dev/video*               0660   system     camera
 /dev/dma_heap/dsp         0660   mediacodec      mediacodec
