diff --git a/imx8m/lec_imx8mp/fstab.nxp b/imx8m/lec_imx8mp/fstab.nxp
index bc5e8ac7..df69d5a1 100644
--- a/imx8m/lec_imx8mp/fstab.nxp
+++ b/imx8m/lec_imx8mp/fstab.nxp
@@ -4,6 +4,7 @@
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
 
 /devices/platform/soc@0/32f10108.usb/38200000.dwc3/* auto auto defaults voldmanaged=usb:auto
+/devices/platform/soc@0/33800000.pcie/* auto auto defaults voldmanaged=pcie:auto
 /dev/block/by-name/userdata    /data        ext4    nosuid,nodev,nodiratime,noatime,noauto_da_alloc,errors=panic    latemount,wait,formattable,quota,fileencryption=aes-256-xts:aes-256-cts:v2,keydirectory=/metadata/vold/metadata_encryption,reservedsize=128M,checkpoint=block
 /dev/block/by-name/metadata    /metadata    ext4    noatime,nosuid,nodev,discard,sync                                                wait,formattable,first_stage_mount,check
 /dev/block/by-name/misc        /misc        emmc    defaults                                                                         defaults
