commit 0601709e2f0ecf758d2e81321f50ddd74eb6cb45
Author: Veera H <veera.h@adlinktech.com>
Date:   Fri Jun 9 16:03:18 2023 +0530

    testing

diff --git a/imx8m/lec_imx8mp/UbootKernelBoardConfig.mk b/imx8m/lec_imx8mp/UbootKernelBoardConfig.mk
index 44768d18..0f04a11c 100644
--- a/imx8m/lec_imx8mp/UbootKernelBoardConfig.mk
+++ b/imx8m/lec_imx8mp/UbootKernelBoardConfig.mk
@@ -3,12 +3,19 @@ TARGET_BOOTLOADER_POSTFIX := bin
 UBOOT_POST_PROCESS := true
 
 # u-boot target
-TARGET_BOOTLOADER_CONFIG := imx8mp:imx8mp_evk_android_defconfig
+#TARGET_BOOTLOADER_CONFIG := imx8mp:imx8mp_evk_android_defconfig
+TARGET_BOOTLOADER_CONFIG := imx8mp:adlink_lec8mp_android_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-2gk:adlink_lec8mp_2gk_android_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-4gb:adlink_lec8mp_4gb_android_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-8gb:adlink_lec8mp_8gb_android_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-trusty:imx8mp_evk_android_trusty_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-secure-unlock:imx8mp_evk_android_trusty_secure_unlock_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-dual:imx8mp_evk_android_dual_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-dual:imx8mp_evk_android_trusty_dual_defconfig
-#TARGET_BOOTLOADER_CONFIG += imx8mp-evk-uuu:imx8mp_evk_android_uuu_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-2gb-uuu:adlink_lec8mp_android_uuu_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-2gk-uuu:adlink_lec8mp_2gk_android_uuu_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-4gb-uuu:adlink_lec8mp_4gb_android_uuu_defconfig
+TARGET_BOOTLOADER_CONFIG += imx8mp-8gb-uuu:adlink_lec8mp_8gb_android_uuu_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-powersave:imx8mp_evk_android_powersave_defconfig
 #TARGET_BOOTLOADER_CONFIG += imx8mp-trusty-powersave:imx8mp_evk_android_trusty_powersave_defconfig
 
diff --git a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-nodpi/default_wallpaper.png b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-nodpi/default_wallpaper.png
index e69de29b..9c41bcba 100644
Binary files a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-nodpi/default_wallpaper.png and b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-nodpi/default_wallpaper.png differ
diff --git a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw600dp-nodpi/default_wallpaper.png b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw600dp-nodpi/default_wallpaper.png
index e69de29b..9c41bcba 100644
Binary files a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw600dp-nodpi/default_wallpaper.png and b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw600dp-nodpi/default_wallpaper.png differ
diff --git a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.png b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.png
index e69de29b..9c41bcba 100644
Binary files a/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.png and b/imx8m/lec_imx8mp/overlay/frameworks/base/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.png differ
