commit cd1624c8aeca49907a9d80199a27ad8af6212dea
Author: Veera H <veera.h@adlinktech.com>
Date:   Thu Jul 27 13:04:11 2023 +0530

    uuu_tool_option_pcf8563_rtc

diff --git a/common/tools/uuu_imx_android_flash.sh b/common/tools/uuu_imx_android_flash.sh
index 545b795a..ff10e021 100755
--- a/common/tools/uuu_imx_android_flash.sh
+++ b/common/tools/uuu_imx_android_flash.sh
@@ -377,7 +377,7 @@ usb_paths=""
 imx8mm_uboot_feature=(dual trusty-dual 4g-evk-uuu 4g ddr4-evk-uuu ddr4 evk-uuu trusty-4g trusty-secure-unlock trusty)
 imx8mn_uboot_feature=(dual trusty-dual evk-uuu trusty-secure-unlock trusty ddr4-evk-uuu ddr4)
 imx8mq_uboot_feature=(dual trusty-dual evk-uuu trusty-secure-unlock trusty)
-imx8mp_uboot_feature=(dual trusty-dual evk-uuu trusty-secure-unlock trusty powersave trusty-powersave)
+imx8mp_uboot_feature=(dual trusty-dual evk-uuu trusty-secure-unlock trusty powersave trusty-powersave 2gb 4gb 8gb)
 imx8ulp_uboot_feature=(dual trusty-dual evk-uuu trusty-secure-unlock trusty)
 imx8qxp_uboot_feature=(mek-uuu trusty-secure-unlock trusty secure-unlock c0 trusty-c0 mek-c0-uuu)
 imx8qm_uboot_feature=(mek-uuu trusty-secure-unlock trusty secure-unlock md hdmi xen)
diff --git a/imx8m/lec_imx8mp/SharedBoardConfig.mk b/imx8m/lec_imx8mp/SharedBoardConfig.mk
index a9d4ae4c..e1da3e86 100644
--- a/imx8m/lec_imx8mp/SharedBoardConfig.mk
+++ b/imx8m/lec_imx8mp/SharedBoardConfig.mk
@@ -7,6 +7,7 @@ IMX8MP_USES_GKI := true
 BOARD_VENDOR_KERNEL_MODULES += \
     $(TARGET_OUT_INTERMEDIATES)/MXMWIFI_OBJ/mlan.ko \
     $(KERNEL_OUT)/drivers/bluetooth/mbtex_8997/bt8xxx.ko \
+    $(KERNEL_OUT)/drivers/rtc/rtc-pcf8563.ko \
     $(TARGET_OUT_INTERMEDIATES)/MXMWIFI_OBJ/moal.ko
 
 # CONFIG_IMX_SDMA: imx-sdma.ko, sdma used for audio
