commit 8c198e50f382272b834083884691f1d236cb5bbb
Author: Veera H <veera.h@adlinktech.com>
Date:   Wed Nov 15 20:05:37 2023 +0530

    SD Card support on eMMC boot

diff --git a/imx8m/lec_imx8mp/fstab.nxp b/imx8m/lec_imx8mp/fstab.nxp
index df69d5a1..71bacb50 100644
--- a/imx8m/lec_imx8mp/fstab.nxp
+++ b/imx8m/lec_imx8mp/fstab.nxp
@@ -10,3 +10,6 @@
 /dev/block/by-name/misc        /misc        emmc    defaults                                                                         defaults
 /dev/block/zram0 none swap defaults zramsize=629145600
 
+/devices/platform/soc@0/30800000.bus/30b50000.mmc/mmc_host/* auto auto defaults voldmanaged=sdcard1:auto,encryptable=userdata
+
+
