diff --git a/arch/arm64/boot/dts/freescale/lec_imx8mp.dts b/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
index 5af5f1182135..87629ee79200 100755
--- a/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
+++ b/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
@@ -222,6 +222,33 @@ lvds_backlight: lvds_backlight {
 	};
 };
 
+
+&clk {
+	init-on-array = <IMX8MP_CLK_HSIO_ROOT>;
+	clocks = <&osc_32k>, <&osc_24m>, <&clk_ext1>, <&clk_ext2>,
+		 <&clk_ext3>, <&clk_ext4>, <&sai1_mclk>, <&sai2_mclk>,
+		 <&sai3_mclk>, <&sai5_mclk>, <&sai6_mclk>, <&sai7_mclk>;
+	clock-names = "osc_32k", "osc_24m", "clk_ext1", "clk_ext2",
+		      "clk_ext3", "clk_ext4", "sai1_mclk", "sai2_mclk",
+		      "sai3_mclk", "sai5_mclk", "sai6_mclk", "sai7_mclk";
+	assigned-clocks = <&clk IMX8MP_CLK_A53_SRC>,
+			  <&clk IMX8MP_CLK_A53_CORE>,
+			  <&clk IMX8MP_CLK_NOC>,
+			  <&clk IMX8MP_CLK_NOC_IO>,
+			  <&clk IMX8MP_CLK_GIC>,
+			  <&clk IMX8MP_CLK_AUDIO_AHB>,
+			  <&clk IMX8MP_CLK_AUDIO_AXI_SRC>,
+			  <&clk IMX8MP_AUDIO_PLL1>,
+			  <&clk IMX8MP_AUDIO_PLL2>,
+			  <&clk IMX8MP_VIDEO_PLL1>,
+			  <&clk IMX8MP_CLK_SAI1>,
+			  <&clk IMX8MP_CLK_SAI2>,
+			  <&clk IMX8MP_CLK_SAI3>,
+			  <&clk IMX8MP_CLK_SAI5>,
+			  <&clk IMX8MP_CLK_SAI6>,
+			  <&clk IMX8MP_CLK_SAI7>;
+};
+
 &A53_0 {
 	cpu-supply = <&buck2>;
 };
@@ -295,38 +322,37 @@ spidev1: spi@0 {
 
 /* eth1 */
 &eqos {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_eqos>;
-	phy-mode = "rgmii-id";
-	phy-handle = <&dp83867_1>;
-	snps,rmii_refclk_ext;
-	//phy-supply = <&dp83867_2v5>;
-	status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_eqos>;
+        phy-mode = "rgmii-id";
+        phy-handle = <&dp83867_1>;
+        //phy-supply = <&dp83867_2v5>;
+        status = "okay";
 
-	mdio {
-		compatible = "snps,dwmac-mdio";
-		#address-cells = <1>;
-		#size-cells = <0>;
-		/* TODO : RESET needed
-		 * TODO : IRQ   needed
-		 */
-		dp83867_1: ethernet-phy@0 {
-			/* RGMII-1 */
-			compatible = "ethernet-phy-id2000.a231"; 
-			reg = <0>;
+        mdio {
+                compatible = "snps,dwmac-mdio";
+                #address-cells = <1>;
+                #size-cells = <0>;
+                /* TODO : RESET needed
+                 * TODO : IRQ   needed
+                 */
+                dp83867_1: ethernet-phy@0 {
+                        /* RGMII-1 */
+                        compatible = "ethernet-phy-id2000.a231";
+                        reg = <0>;
 
-			ti,rx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
-			ti,tx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
-			ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+                        ti,rx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
+                        ti,tx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
+                        ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+                        interrupt-parent = <&gpio4>;
+                        interrupts = <21 IRQ_TYPE_EDGE_FALLING>;
+                        reset-gpio = <&gpio4 22 GPIO_ACTIVE_LOW>;
 
-			ti,min-output-impedance;
-			ti,dp83867-rxctrl-strap-quirk;
+                        ti,min-output-impedance;
+                        ti,dp83867-rxctrl-strap-quirk;
 
-			interrupt-parent = <&gpio4>;
-			interrupts = <21 IRQ_TYPE_EDGE_FALLING>;
-			reset-gpio = <&gpio4 22 GPIO_ACTIVE_LOW>;
-		};
-	};
+                };
+        };
 };
 
 /* eth0 */
@@ -340,7 +366,6 @@ &fec {
 	status = "okay";
 
 	mdio {
-		compatible = "snps,dwmac-mdio";
 		#address-cells = <1>;
 		#size-cells = <0>;
 
@@ -353,7 +378,6 @@ dp83867_0: ethernet-phy@1 {
 			interrupt-parent = <&gpio4>;
 			interrupts = <3 IRQ_TYPE_EDGE_FALLING>;
 			reset-gpio = <&gpio4 4 GPIO_ACTIVE_LOW>;
-			micrel,led-mode = <0>;
 			ti,rx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
 			ti,tx-internal-delay = <DP83867_RGMIIDCTL_1_75_NS>;
 			ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
@@ -364,22 +388,6 @@ dp83867_0: ethernet-phy@1 {
 	};
 };
 
-&flexspi {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexspi0>;
-	status = "okay";
-
-	flash0: mt25qu256aba@0 {
-		reg = <0>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		spi-max-frequency = <80000000>;
-		spi-tx-bus-width = <1>;
-		spi-rx-bus-width = <4>;
-	};
-};
-
 &flexcan1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan1>;
@@ -393,7 +401,7 @@ &flexcan2 {
 };
 
 &i2c1 {
-	clock-frequency = <400000>;
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
@@ -548,7 +556,7 @@ gpio8: i2c2_gpioext0@3e {
 };
 
 &i2c3 {
-	clock-frequency = <400000>;
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
@@ -1184,17 +1192,6 @@ MX8MP_IOMUXC_SAI2_MCLK__GPIO4_IO27	0x154	/* CAN2_STBY */
 		>;
 	};
 
-	pinctrl_flexspi0: flexspi0grp {
-		fsl,pins = <
-			MX8MP_IOMUXC_NAND_ALE__FLEXSPI_A_SCLK           0x1c2
-			MX8MP_IOMUXC_NAND_CE0_B__FLEXSPI_A_SS0_B        0x82
-			MX8MP_IOMUXC_NAND_DATA00__FLEXSPI_A_DATA00      0x82
-			MX8MP_IOMUXC_NAND_DATA01__FLEXSPI_A_DATA01      0x82
-			MX8MP_IOMUXC_NAND_DATA02__FLEXSPI_A_DATA02      0x82
-			MX8MP_IOMUXC_NAND_DATA03__FLEXSPI_A_DATA03      0x82
-		>;
-	};
-
 	pinctrl_gpio_led: gpioledgrp {
 		fsl,pins = <
 			MX8MP_IOMUXC_NAND_READY_B__GPIO3_IO16	0x19
@@ -1322,10 +1319,8 @@ MX8MP_IOMUXC_UART2_TXD__UART2_DCE_TX	0x49
 
 	pinctrl_uart3: uart3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_ECSPI1_SCLK__UART3_DCE_RX		0x140
-			MX8MP_IOMUXC_ECSPI1_MOSI__UART3_DCE_TX		0x140
-			MX8MP_IOMUXC_ECSPI1_SS0__UART3_DCE_RTS		0x140
-			MX8MP_IOMUXC_ECSPI1_MISO__UART3_DCE_CTS		0x140
+			MX8MP_IOMUXC_NAND_CE0_B__UART3_DCE_TX	0x140
+			MX8MP_IOMUXC_NAND_ALE__UART3_DCE_RX	0x140
 		>;
 	};
 
diff --git a/arch/arm64/configs/lec_imx8mp_gki.fragment b/arch/arm64/configs/lec_imx8mp_gki.fragment
index bd6adddd96f9..e31e3abd4586 100644
--- a/arch/arm64/configs/lec_imx8mp_gki.fragment
+++ b/arch/arm64/configs/lec_imx8mp_gki.fragment
@@ -173,3 +173,5 @@ CONFIG_USB_CDNS3_GADGET=y
 CONFIG_USB_CDNS3_HOST=y
 CONFIG_USB_MUSB_HDRC=y
 CONFIG_USB_DWC3_DUAL_ROLE=y
+CONFIG_MVMDIO=y
+CONFIG_TSN=y
