diff --git a/arch/arm64/configs/lec_imx8mp_gki.fragment b/arch/arm64/configs/lec_imx8mp_gki.fragment
index e31e3abd4586..2706e5bc14ab 100644
--- a/arch/arm64/configs/lec_imx8mp_gki.fragment
+++ b/arch/arm64/configs/lec_imx8mp_gki.fragment
@@ -175,3 +175,7 @@ CONFIG_USB_MUSB_HDRC=y
 CONFIG_USB_DWC3_DUAL_ROLE=y
 CONFIG_MVMDIO=y
 CONFIG_TSN=y
+CONFIG_BLK_DEV_NVME=y
+CONFIG_NVME_FC=y
+CONFIG_NVME_TCP=y
+CONFIG_NVME_TARGET=y
