commit 9841c1b53ba60958e04bba1ea9f33f36dc2950d3
Author: Veera H <veera.h@adlinktech.com>
Date:   Thu Jul 27 13:07:38 2023 +0530

    rtc-pcf8563-adlbmc

diff --git a/arch/arm64/boot/dts/freescale/lec_imx8mp.dts b/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
index 974509064b8b..1b80622c15a6 100755
--- a/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
+++ b/arch/arm64/boot/dts/freescale/lec_imx8mp.dts
@@ -821,13 +821,13 @@ pcf8563: rtc@51 {
         };
 
         /* Lite-BMC U1101 */
-        /*adl-bmc@28 {
+        adl-bmc@28 {
                 compatible = "adl-bmc";
                 reg = <0x28>;
                 #address-cells = <1>;
                 #size-cells = <0>;
                 clocks = <&pcf8563>;
-        };*/
+        };
 };
 
 &irqsteer_hdmi {
diff --git a/arch/arm64/configs/lec_imx8mp_gki.fragment b/arch/arm64/configs/lec_imx8mp_gki.fragment
index 1597626040e9..5768e7e32a25 100644
--- a/arch/arm64/configs/lec_imx8mp_gki.fragment
+++ b/arch/arm64/configs/lec_imx8mp_gki.fragment
@@ -180,3 +180,4 @@ CONFIG_NVME_FC=y
 CONFIG_NVME_TCP=y
 CONFIG_NVME_TARGET=y
 CONFIG_DRM_PANEL_AUO_B101UAN1V7=y
+CONFIG_RTC_DRV_PCF8563=m
