commit 427f998693f0a441f71a1e5eb024242741c8b827
Author: adlinkshiva <shivashankar.thiruchitrambalam@adlinktech.com>
Date:   Sun Apr 7 20:45:45 2024 +0530

    0017-panel_fix

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index ce73d74e1e47..5d6c37ccd492 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -5393,9 +5393,9 @@ static int panel_simple_dsi_probe(struct mipi_dsi_device *dsi)
 	struct device_node *np;
 	const struct panel_desc_dsi *desc;
 	const struct of_device_id *id;
-	u32 dsi_flags;
-	u32 dsi_format;
-	u32 dsi_lanes;
+	u32 dsi_flags = 0;
+	u32 dsi_format = 0;
+	u32 dsi_lanes = 0;
 	int err;
 
 	np = dsi->dev.of_node;
