diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index 4569bceff9..e75dd62c9d 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -363,12 +363,35 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += persist.traced.enable=1
 PRODUCT_PACKAGES_DEBUG := \
     adb_keys \
     arping \
+    adlink_pwm \
+    tinycap2 \
+    tinymix2 \
+    tinypcminfo2 \
+    tinyplay2 \
     gdbserver \
     idlcli \
     init-debug.rc \
     iotop \
+    alsa_aplay \
+    alsa_ctl \
+    alsa_ucm \
+    alsa_amixer \
+    candump \
+    cansend \
+    cansequence \
+    hcidump \
+    hciconfig \
+    hcitool \
+    hciattach \
     iperf3 \
     iw \
+    vim \
+    wpa_passphrase \
+    iwconfig \
+    dhclient \
+    dhtool \
+    dhcpclient \
+    dhcptool \
     logpersist.start \
     logtagd.rc \
     procrank \
@@ -383,6 +406,12 @@ PRODUCT_PACKAGES_DEBUG := \
     tracepath \
     tracepath6 \
     traceroute6 \
+    fswebcam \
+    spidev_test \
+    wpa_supplicant \
+    wpa_cli \
+    libwpa_client \
+    libwpa_hidl \
     unwind_info \
     unwind_reg_info \
     unwind_symbols \
